method/link.js

```diff
@@ -15,25 +75,19 @@ module.exports = function Link (ssb) {
     if (child && !isGroup(child)) return cb(new Error(`findLinks expected a groupId for child, got ${child} instead.`))
     if (!parent && !child) return cb(new Error('findLinks expects a parent or child id to be given'))
 
-    const query = [{
-      $filter: {
-        value: {
-          content: {
-            type,
-            ...opts,
-            tangles: {
-              link: { root: null, previous: null }
-            }
-          }
-        }
-      }
-    }]
-
     pull(
-      // NOTE: using ssb-query instead of ssb-backlinks
-      // because the backlinks query will get EVERY message which contains the groupId in it, which will be a LOT for a group
-      // then filters that massive amount down to the ones which have the dest in the right place
-      ssb.query.read({ query }),
+      ssb.db.query(
+        where(
+          and(
+            dbType(type),
+            parent && equal(seekParent, parent, { indexType: 'parent', prefix: true }),
+            child && equal(seekChild, child, { indexType: 'child', prefix: true }),
```

Comment:
```suggestion
            parent && equal(seekParent, parent, { indexType: 'linkParent', prefix: true }),
            child && equal(seekChild, child, { indexType: 'linkChild', prefix: true }),
```

going for more specific  index names by default.
BTW I am new the the `prefix: true` thing. It seems like this is the right choice here, did you ask any of the butt-db elders about this?

Reply:
will change the index names.

didn't ask about prefix, i was hoping you were familiar :grin: i got the impression from the db2 or jitdb readme that prefix was good here but i wasn't 100%. what do you say @staltz ?

Reply:
changed index names